polynote-kernel/src/main/scala/polynote/kernel/remote/RemoteKernel.scala

```diff
@@ -312,22 +305,30 @@ object RemoteKernelClient extends polynote.app.App {
     interpreterFactories: Map[String, List[Interpreter.Factory]],
     kernelFactory: Factory.Service,
     polynoteConfig: PolynoteConfig
-  ): Task[KernelEnvironment] = {
-    val publishStatus = publishResponse.contramap(KernelStatusResponse.apply)
-    TaskManager(publishStatus).map {
-      taskManager => KernelEnvironment(
-        currentNotebook,
-        reqId,
-        publishResponse,
-        interpreterFactories,
-        kernelFactory: Factory.Service,
-        polynoteConfig: PolynoteConfig,
-        taskManager,
-        publishStatus
-      )
+  ): Layer[Throwable, GlobalEnv with CellEnv with PublishRemoteResponse] = {
+    val publishStatus = PublishStatus.layer(publishResponse.contramap(KernelStatusResponse.apply))
+    val publishRemote: Layer[Nothing, PublishRemoteResponse] = ZLayer.succeed(publishResponse)
+
+    Config.layer(polynoteConfig) ++
+      CurrentNotebook.layer(currentNotebook) ++
+      Interpreter.Factories.layer(interpreterFactories) ++
+      ZLayer.succeed(kernelFactory) ++
+      publishStatus ++
+      (publishStatus >>> TaskManager.layer) ++
```

Comment:
Wondering why you have `(publishStatus >>> TaskManager.layer)` instead of something simpler(?) like 

```
val statusUpdates = publishResponse.contramap(KernelStatusResponse.apply)
val publishStatus = PublishStatus.layer(statusUpdates)
val taskManager = ZLayer.succeed(statusUpdates)

...add up all them layers...
```

I understand why we have `(publishRemote >>> mkResultPublisher(reqId))`, because it's used in multiple places and other places don't have easy, non-environment access to `statusUpdates`. 

So maybe I'm just being pedantic and you had some other access to `TaskManager.layer` which was lost in some refactor? You can tell me if I'm being pedantic 😄 I'm probably just being pedantic.